tools/lsp/ui/main.slint

```diff
@@ -38,113 +38,131 @@ export component PreviewUi inherits Window {
-                        mouse-cursor: crosshair;
-                        enabled <=> root.design-mode;
+                        resize(w, h) => {
+                            i-preview-area-container.width = clamp(w, i-preview-area-container.min-width, i-preview-area-container.max-width);
+                            i-preview-area-container.height = clamp(h, i-preview-area-container.min-height, i-preview-area-container.max-height);
+                        }
 
-                        x: i-preview-area-container.x;
-                        y: i-preview-area-container.y;
                         width: i-preview-area-container.width;
                         height: i-preview-area-container.height;
-                    }
 
-                    i-selection-display-area := Rectangle {
-                        x: i-preview-area-container.x;
-                        y: i-preview-area-container.y;
-                        width: i-preview-area-container.width;
-                        height: i-preview-area-container.height;
+                        i-preview-area-container := ComponentContainer {
+
+                            property <bool> is-resizable: (self.min-width != self.max-width && self.min-height != self.max-height) && self.has-component;
+
+                            component-factory <=> root.preview-area;
+
+                            // The width and the height can't depend on the layout info of the inner item otherwise this would
+                            // cause a recursion if this happens (#3989)
+                            // Instead, we use a init function to initialize
+                            width: 0px;
+                            height: 0px;
+                            init => {
+                                self.width = max(self.preferred-width, self.min-width);
+                                self.height = max(self.preferred-height, self.min-height);
+                            }
+                        }
+
+                        // Also make a condition that abuses the fact that the init callback
+                        // is called everytime the condition is dirty, to make sure that the size
+                        // is within the bounds.
+                        // Querty the preview-area to make sure this is evaluated when it changes
+                        if i-preview-area-container.has-component && root.preview-area == i-preview-area-container.component-factory : Rectangle {
+                            init => {
+                                i-preview-area-container.width = clamp(i-preview-area-container.width, i-preview-area-container.min-width, i-preview-area-container.max-width);
+                                i-preview-area-container.height = clamp(i-preview-area-container.height, i-preview-area-container.min-height, i-preview-area-container.max-height);
+                            }
+                        }
```

Comment:
I guess in the future we can do this with a changed callback? (`property <bool> condition: ...; changed condition =>  { ... }`)

Reply:
yes, if we had changed callback we couldn't need that.